build.sbt

```diff
@@ -101,11 +102,14 @@ lazy val wartRemoverSettings = {
   wartremoverErrors in(Compile, compile) ++= Warts.allBut(excludedWarts: _*)
 }
 
+lazy val catsSettings = scalacOptions += "-Ypartial-unification"
```

Comment:
I added this (and `scalaVersion := "2.11.12"`) in to avoid the need for type aliases to get the `.mapN` in the `Account` and `Transaction` validation to compile.

It did work, however IntelliJ still highlighed the `.mapN` calls in red without the type aliases so I left the type aliases in place. That means that this `-Ypartial-unification` is not strictly necessary for the code in this PR to compile, however I left in in place because it is recommended by https://typelevel.org/cats/

Reply:
@frankoid thanks for that - creating type aliases here always gnawed at my side so it's good to be able to not use it. It's slightly annoying that intelliJ can't figure it out and shows an error though, but overall it's cleaner being able to not use the type aliases.